src/test/java/org/carlspring/cloud/storage/s3fs/path/S3PathTest.java

```diff
@@ -147,17 +136,6 @@ void registerThrowException()
         assertNotNull(exception);
     }
 
-    @Test
-    void registerWithEventsAndModifierThrowException()
```

Comment:
This should be reverted - it appears to have been intended to cover the case when you have all 3 arguments in the `.register()`, but was implemented incorrectly. The proper implementation would be:

```

    /**
     * This covers {@link S3Path#register (WatchService watcher, WatchEvent.Kind<?>[] events, WatchEvent.Modifier... modifiers)
     */
    @Test
    void registerWatcherWithEventsAndModifierShouldThrowUnsupportedOperationException()
    {
        // We're expecting an exception here to be thrown
        Exception exception = assertThrows(UnsupportedOperationException.class, () -> {
            forPath("file1").register(null, null, null);
        });

        assertNotNull(exception);
    }
```